compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1315,6 +1320,11 @@ private BType inferTableMemberType(List<BType> memTypes, BLangTableConstructorEx
         return createTableConstraintRecordType(inferredFields, restFieldTypes, tableConstructorExpr.pos, data);
     }
 
+    public boolean isSameTableType(BTableType source, BTableType target) {
```

Comment:
Can you check whether `Types.isSameType()` can be used instead?

Reply:
Sure

Reply:
We can not use that, since it check for memory equality. Here `tableType` is new type created to get the key specifier which can be available in the rhs or lhs.
https://github.com/ballerina-platform/ballerina-lang/blob/ac3f47fa8fc9918cffc70acf5bb232cc4627c63e/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java#L2807

Reply:
```suggestion
    private boolean isSameTableType(BTableType source, BTableType target) {
```